Cython/Compiler/TypeStubGenerator.py

```diff
@@ -0,0 +1,485 @@
+        super(DeclarationWriter, self).__init__()
+        self.context = context
+        self.module_name = ""
+        self.class_func_count = 0 
+        # TODO Maybe have arguments 
+        # for how shadow.pyi is implemented be passed into here 
+        # if we want them to be even imported at 
+        # all and if so C Variables should be translated as such...
+    
+        # TODO in the future allow context variables to be passed so that a directive can be used to 
+        # pass along docstring infromation so tools such as sphinx can generate clean documentation...
+
+        self.translation_table:dict[str,str] = {}
+        """Used as an eternal resource for translating ctype declarations into python-types"""
+
+        self.use_typing : bool = False
+        """if true we must import typing's generator typehint..."""
+
+
+    def _visitchildren_indented(self, node):
+        self.indent()
+        self.visitchildren(node)
+        self.dedent()
+    
+    def translate_pyrex_type(self, ctype:PyrexTypes.PyrexType):
+        # TODO implement Pyrex to cython shadow typehints converter...
+        
+        if isinstance(ctype, PyrexTypes.CIntType):
+            return "int"
+
+        elif isinstance(ctype, PyrexTypes.CFloatType):
+            return "float"
+    
+        elif isinstance(ctype,PyrexTypes.PyObjectType):
+            py_name = ctype.py_type_name()
+            if py_name:
+                # Try returning the python type name and put it into quotes 
+                # incase the object has not been registered yet....
+                return py_name if py_name in self.stack else f"{py_name!r}"
+
+        return 'object'
+
+
+    # Instead of doing it into C, we're doing it backwards...
+    def translate_base_type_to_py(
+        self,
+        base:CSimpleBaseTypeNode
+        ):
+
+        # Try checking our table first...
+        if self.translation_table.get(base.name):
+            return self.translation_table[base.name]
+
+        elif base.name == "object":
+            return "object"
+
+        elif base.name in ("unicode","basestring"):
+            return "str"
+
+        elif not base.is_basic_c_type:
+            # Likely that it's already a python object that's being handled...
+            # execpt for basestring and unicode...
+            return base.name 
+
+        elif base.name == "bint":
+            return "bool"
+
+        ctype = PyrexTypes.simple_c_type(base.signed, base.longness, base.name) # type: ignore
+        return self.translate_pyrex_type(ctype)
+
+    def emptyline(self):
+        self.result.putline("")
+
+    def visit_ModuleNode(self, node: ModuleNode):
+        # We need to extract the name to write our pyi file down...
+        if node.directives['write_stub_file']:
+            result = self.write(node, True)
+            # TODO See if we should or shouldn't report back about our stubfiles being written off....
+            print(f"Writing {result}.pyi ...") 
```

Comment:
We can't assume that Cython runs of a version of Python that supports fstrings yet